### PR TITLE
[JIT] Fork/Join inline docs

### PR DESCRIPTION
**Pull request by eellison**

### CODE REVIEW

File: torch/jit/__init__.py [hunk 1]
```diff
@@ -789,6 +789,103 @@ def wrap_check_inputs(check_inputs):
...
+    Arguments:
+        func (callable or torch.nn.Module):  A Python function or `torch.nn.Module`
+            that will be invoked. If executed in TorchScript, it will execute asynchronously,
+            otherwise it will not. Traced invocations of fork will be captured in the IR.
+
+        *args, **kwargs: arguments to invoke `func` with.
+
+    Returns:
+        `torch.jit.Future[T]`: a reference to the execution of `func`. The value `T`
+        can only be accessed by forcing copmletion of `func` through `torch.jit.wait`.
```
Comment (jamesr66a) [inline torch/jit/__init__.py:822]: `completion`

File: torch/jit/__init__.py [hunk 2]
```diff
@@ -789,6 +789,103 @@ def wrap_check_inputs(check_inputs):
...
+
+        import torch
+        from torch import Tensor
+
+        def foo(a : Tensor, b : int) -> Tensor:
+            return a + b
+
+        def bar(a):
+            fut : torch.jit.Future[Tensor] = torch.jit._fork(foo, a, b=2)
+            return torch.jit._wait(fut)
```
Comment (jamesr66a) [inline torch/jit/__init__.py:836]: Might want to update these to remove the leading underscore on the fn's

File: torch/jit/__init__.py [hunk 3]
```diff
@@ -789,6 +789,103 @@ def wrap_check_inputs(check_inputs):
...
+                return a + b
+
+        class Mod(torch.nn.Module):
+            def __init__(self):
+                super(self).__init__()
+                self.mod = SubMod()
+
+            def forward(self, input):
+                fut = torch.jit._fork(self.mod, a, b=2)
+                return torch.jit._wait(fut)
```
Comment (jamesr66a) [inline torch/jit/__init__.py:866]: Same here, leading underscores

**Review by jamesr66a — APPROVED** [review]
Nice!

File: torch/jit/__init__.py [hunk 4]
```diff
@@ -789,6 +789,103 @@ def wrap_check_inputs(check_inputs):
 
     return [{'forward' : c} for c in check_inputs]
 
+
+def fork(func,
+          *args,
+          **kwargs):
+    """
+    Creates an asynchronous task executing `func` and return a reference
+    to the execution of this function. `fork` will return immediately,
```
Comment (jamesr66a) [inline torch/jit/__init__.py:798]: In my doc, I referred to the return value as "a reference to the value of the result of this execution", just to emphasize that it's a value contained in the Future

**Comment by seemethere** [comment]
hey @eellison! Just a heads up, we're cutting a release branch for 1.6.0 tomorrow and would love to have this feature in for that!

**Comment by eellison** [comment]
> hey @eellison! Just a heads up, we're cutting a release branch for 1.6.0 tomorrow and would love to have this feature in for that!

yep, working on landing it right now

**Comment by ilia-cher** [comment]
it seems to break 
pytorch_python_doc_push 
https://app.circleci.com/jobs/github/pytorch/pytorch/5967777
https://ezyang.github.io/pytorch-ci-hud/build/pytorch-master

**Comment by ilia-cher** [comment]
@eellison

**Comment by eellison** [comment]
it's kind of tough to fix the test when there are no instructions to reproduce locally. the failing test script says `# =================== The following code **should** be executed inside Docker container ===================`

**Comment by zhaojuanmao** [comment]
reverted as it broke pytorch_python_doc_push 

https://app.circleci.com/pipelines/github/pytorch/pytorch/184261/workflows/6ff19265-3b1c-43c3-896f-d1b0803fcb8f/jobs/5967777/steps

**Pull request closed without merging**